hca/upload/media_type.py

```diff
@@ -0,0 +1,90 @@
+import magic
+import os
+import re
+
+
+class MediaType:
+
+    RFC7320_TOKEN_CHARSET = "!#$%&'*+-.^_`|~\w"
+    DCP_METADATA_JSON_FILES = ('assay', 'project', 'sample')
+
+    @classmethod
+    def from_string(cls, media_type):
+        split_on_semicolons = media_type.split(';')
```

Comment:
if you have semicolons in the media type k/v pairs sections, will this break?

Reply:
It is valid to have multiple sets of k/v params separated by semicolons.  This works in that case.  I have added a test case for that.

For the case where a semicolon is inside a quoted string, yes this will break in that case.  Given that the use of parameters is pretty obscure anyway, I'm not sure it's worth the effort of spinning up a full blown parser which I think is what you would need to do to handle this case.  Unless you know a shortcut?

Reply:
this syntax looks remarkably like the cookie header syntax -- pairs separated by semicolons, values can be quoted.  https://docs.python.org/2/library/cookie.html

Reply:
if the cookie library doesn't work for your purposes, and you're going to assume that ';' is an uncommon value, i would suggest that you flag it when someone attempts to set a value with a ';'.

Reply:
Makes sense.  Let me look at the cookie stuff.